api/main.go

```diff
@@ -13,7 +14,7 @@ var (
 
 // API Errors
 var (
-	ErrAuthenticationAlreadySet = errors.New("the authentication function has already been set")
+	ErrAuthenticationAlreadySet = errors.New("the authentication function has already been set (or must be set earlier)")
```

Comment:
Create a new error for that. Error semantics should be clear and represent a single failure. Testing against ErrAuthenticationAlreadySet is useless now because one does not know the actual error.